android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java

```diff
@@ -0,0 +1,79 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.util.Pair;
+
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.data.extras.EventDates;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.utils.DateConverter;
+import org.threeten.bp.format.DateTimeParseException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class ScheduleFragmentViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<List<Pair<String,String>>> eventDatesStringLiveData;
+    private MutableLiveData<List<Track>> tracksLiveData;
+    private RealmResults<Track> tracksRealmResults;
+    private RealmResults<EventDates> eventDatesRealmResults;
+
+
+    public ScheduleFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<Pair<String,String>>> getEventDatesString() {
+        if (eventDatesStringLiveData == null) {
+            eventDatesStringLiveData = new MutableLiveData<>();
+            eventDatesRealmResults = realmRepo.getEventDates();
+            eventDatesRealmResults.addChangeListener((eventDates, orderedCollectionChangeSet) -> {
+                        List<Pair<String,String>> eventDatesString = new ArrayList<>();
+                        int eventDays = eventDates.size();
+                        for (int i = 0; i < eventDays; i++) {
+                            String date = eventDates.get(i).getDate();
+                            try {
+                                String formattedDate = DateConverter.formatDay(date);
+                                eventDatesString.add(new Pair<>(date,formattedDate));
+                            } catch (DateTimeParseException pe) {
+                                Timber.e(pe);
+                                Timber.e("Invalid date %s in database", date);
+                            }
+                        }
+                        eventDatesStringLiveData.setValue(eventDatesString);
+                    });
+        }
+        return eventDatesStringLiveData;
+    }
+
+    public LiveData<List<Track>> getTracks() {
+        if (tracksLiveData == null) {
+            tracksLiveData = new MutableLiveData<>();
+            tracksRealmResults = realmRepo.getTracks();
+            tracksRealmResults.addChangeListener((tracks, orderedCollectionChangeSet) -> {
+                tracksLiveData.setValue(tracks);
+            });
+        }
+        return tracksLiveData;
+    }
+
+    @Override
+    protected void onCleared() {
+        if (eventDatesRealmResults != null) {
+            eventDatesRealmResults.removeAllChangeListeners();
+            eventDatesRealmResults = null;
```

Comment:
No need to set item to null. we are not passing the instance anywhere, and even if we were, then setting it to null wouldn't have made any difference. Java is copy by value. If this pattern was possible, this would have been such an elegant solution to memory leak that we wouldn't have needed architecture components :stuck_out_tongue_winking_eye: